src/setup/setup.ts

```diff
@@ -0,0 +1,17 @@
+type ConfigProps = {
+	apiKey?: string;
```

Comment:
I think the type here for `apiKey` is required right? We can also make use of Typescript's `Partial` or `Required` utility types to slightly tweak the base types. Let me know what you think.

Reply:
you're right, apiKey is the only one that's required. Everything else is optional. I could make a type for apiKey & secret individually to utilize `Partial` AND `Required`. Is that what you were thinking?